source/features/mobile-tabs.css

```diff
@@ -1,85 +1,99 @@
+	:root .UnderlineNav {
 		padding-inline: var(--base-size-16, 16px);
 		padding-block: 5px var(--control-medium-gap, 8px);
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-body {
+	:root .UnderlineNav-body {
 		align-items: stretch; /* Ensure they have the same height */
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-body > li {
-		flex-shrink: 0;
-		flex-basis: 0;
-	}
-
-	[rgh-mobile-tabs] .UnderlineNav-item {
-		display: flex;
-		justify-content: center;
-		flex-flow: row wrap;
-		padding: 5px;
-		flex-basis: 0;
+	:root .UnderlineNav-item {
+		display: grid;
+		min-width: 50px;
+		grid:
+			'icon counter'
+			'title title';
+		gap: 0 5px;
+		padding: 6px 10px;
 		background: var(
 			--control-transparent-bgColor-hover,
 			var(--color-action-list-item-default-hover-bg)
 		);
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-item:hover {
+	:root .UnderlineNav-item:hover {
 		background: var(
 			--control-transparent-bgColor-hover,
 			var(--color-action-list-item-default-active-bg)
 		); /* From: Counter background color */
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-item.selected::after {
+	:root .UnderlineNav-item.selected::after {
 		/* Reposition the orange line */
 		bottom: -9px !important;
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-item [data-content] {
-		order: 2;
-		flex-basis: 100%;
+	:root .UnderlineNav-item [data-content] {
+		grid-area: title;
 		line-height: 1;
 		margin-top: 5px;
 
-		/* Ensure that there's at least 10px padding, which is usually granted by the empty grid columns */
-		padding-inline: 10px;
-
 		/* font-size: 0; */
 		/* Safari 16.4 color bug #6473 */
 		font-size: 0.6px;
 		color: transparent !important;
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-item [data-content]::before {
+	:root .UnderlineNav-item [data-content]::before {
 		visibility: inherit;
 		height: auto;
 		font-size: 10px;
 		color: var(--fgColor-default, var(--color-fg-default));
 		font-weight: inherit;
 	}
 
-	[rgh-mobile-tabs] [data-content='Pull requests']::before {
+	:root [data-content='Pull requests']::before {
 		content: 'Pulls' !important;
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-item svg {
-		margin-right: 0 !important;
+	:root .UnderlineNav-item svg {
+		margin: 0 !important;
 		grid-area: icon;
-		justify-self: center; /* Improve centering if there's no counter */
+		justify-self: end;
 	}
 
-	[rgh-mobile-tabs] .UnderlineNav-item .Counter {
+	:root .UnderlineNav-item .Counter {
 		grid-area: counter;
 		background: none; /* Counters appears as pills */
-		min-width: 0; /* Pills have a min-width we don't need */
-		padding: 0; /* Pills have padding */
-		text-align: left; /* Pills have centered text */
-		margin-left: 5px;
+		padding: 0;
+		margin: 0;
+		border: none;
+		min-width: min-content;
+		justify-self: start;
+	}
+
+	/* When counter is missing */
+	:root .UnderlineNav-item:not(:has(.Counter)),
+	:root .UnderlineNav-item:has(.Counter:empty) {
+		grid-template-areas:
+			'icon icon'
+			'title title';
+
+		svg {
+			justify-self: center;
+		}
 	}
 }
```

Comment:
I wish there was a way to do this natively in CSS Grid.

In short, the icon should _span_ and _center_ when the counter is missing or empty.

If you have any ideas, please try them and open a PR. I already tried all the "what about X" 😅